core/include/detray/utils/unit_vectors.hpp

```diff
@@ -0,0 +1,80 @@
+/** Detray library, part of the ACTS project (R&D line)
+ *
+ * (c) 2022 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+#pragma once
+
+// Project include(s).
+#include "detray/definitions/qualifiers.hpp"
+
+// Thrust include(s)
+#include <thrust/pair.h>
```

Comment:
I thought it was only for tuple?

Reply:
I think the idea was, to not make a CPU-only build dependent on thrust at all.

Reply:
In that case, I would prefer to have the implementation of `pair` rather than using `std::array` like we do it for tuple. 
But before then, I think we can keep using `thrust::pair` - swapping the type would be very easy

Reply:
OK